### PR TITLE
Implement inventory

### PR DESCRIPTION
**Pull request by bqx**
Also:
* Tweak icons (border, gap) for slot type icons in an empty sheet slot
* Ignore the secondary slot type if its missing from the card type for slot validation
* Crop the active spell icon to remove the border

### CODE REVIEW

File: src/components/ListInventory/index.tsx [hunk 1]
```diff
@@ -1,11 +1,96 @@
...
+                <Icon path={cardData.iconLink} height={36} width={36} />
+                <CostBox cost={cardData.energyCost} />
+                <NameBox>{cardData.name}</NameBox>
+                <AttributesImpactedView attributesImpacted={cardData.attributesImpacted} />
+              </MainContent>
+            </Container>
+          </BaseCharacterSheetSlot>
+        );
+      })}
+      {!!selectedCardForModal && (
```
Comment (rissole) [inline src/components/ListInventory/index.tsx:67]: nit: i dont like the `!!` pattern as it doesnt optimise for readability for others and is less explicit. i prefer  `selectedCardForModal !== null` here

File: src/components/EmptyCharacterSheetSlot/index.tsx [hunk 2]
```diff
@@ -95,7 +96,8 @@ const EmptySlotTypeIcons = ({ slotTypes }: { slotTypes: CardSlotMetadata["slotTy
                 height: "42px",
                 width: "42px",
                 opacity: "70%",
-                borderRadius: "2px",
+                borderRadius: "8px",
+                border: "1px solid black",
```
Comment (rissole) [inline src/components/EmptyCharacterSheetSlot/index.tsx:100]: i think there is too much border radius here (we usually use 2px i think) and the border is a bit too harsh. I would like to try a box shadow type approach as we use in the health/energy/gold icons at the bottom which is a bit more subtle but doesnt block this PR

**Review by rissole — APPROVED** [review]

**Review by ThatRanga — APPROVED** [review]
Not something that needs to be fixed now but found that if you add more and more items into inventory it pushes the status bar at the bottom off the screen. Making it inaccesible.

image

File: src/components/EmptyCharacterSheetSlot/index.tsx (hunk 2 above)
Comment (bqx) [reply src/components/EmptyCharacterSheetSlot/index.tsx:100]: Yep sure, I wanted to try and add more distinction between these empty slots, but don't feel too strongly about this

File: src/components/ListInventory/index.tsx (hunk 1 above)
Comment (bqx) [reply src/components/ListInventory/index.tsx:67]: I don't feel strongly about this, happy to do explicit checks instead. If we wanted we could add a linting rule

**Pull request merged**